src/ipc/server.rs

```diff
@@ -237,3 +226,20 @@ async fn process(ctx: &ClientCtx, request: Request) -> Reply {
 
     Ok(response)
 }
+
+pub fn smithay_window_to_ipc(window: &smithay::desktop::Window) -> niri_ipc::Window {
```

Comment:
Ideally I'd have wanted this function to be a method on either `niri_ipc::Window` or on `smithay::desktop::Window`

However, doing that would either require adding smithay and some other stuff as deps of niri-ipc, or adding an extension trait which would be impl'd on smithay windows. That was extra annoying since methods are being called on the window which requires more work.

For now, this is the cleanest solution I came up with